framework/CodeInterfaces/PHISICS/XSCreator.py

```diff
@@ -0,0 +1,177 @@
+      @ Out, None
+    """
+    rough_string = ET.tostring(elem, 'utf-8')
+    reparsed = minidom.parseString(rough_string)
+    return reparsed.toprettyxml(indent="  ")
+    
+  def generateXML(self,workingDir):
+    """
+      Creates an xml file from the interface.
+      @ In, workingDir, string, absolute path to working directory
+      @ Out, modifiedFile, string, name of the xml file created (under a dummy name)
+    """
+    top = Element('scaling_library')
+    comment = Comment('Generated by rouxpn')
+    top.append(comment)
+    
+    for XS in self.listedDict.iterkeys():
+      for tabulation in self.listedDict.get('XS').iterkeys():
+        topChild = SubElement(top, 'set')
+        tabList, valueList = self.tabMapping(tabulation)
+        if self.booleanTab is True:
+          for i in xrange (0,len(tabList)):
+            tabChild = SubElement(topChild, 'tab', {'name':tabList[i]})
+            tabChild.text = valueList[i]
+        for material in self.listedDict.get('XS').get(tabulation).iterkeys():
+          tabulationChild = SubElement(topChild, 'library', {'lib_name':material.lower()})
+          for isotope in self.listedDict.get('XS').get(tabulation).get(material).iterkeys():
+            for type in self.listedDict.get('XS').get(tabulation).get(material).get(isotope).iterkeys():
+              libraryChild = SubElement(tabulationChild, 'isotope', {'id':isotope.lower(), 'type':type.lower()})
+              for reaction in self.listedDict.get('XS').get(tabulation).get(material).get(isotope).get(type).iterkeys():
+                groupList = []
+                valueList = []
+                count = 0
+                for group,value in self.listedDict.get('XS').get(tabulation).get(material).get(isotope).get(type).get(reaction).iteritems():
+                  count = count + 1
+                  numberOfGroupsPerturbed = len(self.listedDict.get('XS').get(tabulation).get(material).get(isotope).get(type).get(reaction).keys())
+                  groupList.append(group)
+                  valueList.append(value)
+                  group.join(',')
+                  if count == numberOfGroupsPerturbed:
+                    groups = ','.join(str(e) for e in groupList)
+                    values = ','.join(str(f) for f in valueList)
+                    reactionChild = SubElement(libraryChild, reaction.lower(), {'g':groups})
+                    reactionChild.text = values
+    
+    modifiedFile = os.path.join(workingDir,'test.dat')
+    file_obj = open(modifiedFile, 'w')
+    file_obj.write(self.prettify(top))
+    return modifiedFile
+  
+  def cleanEmpty(self,leanDict):
+    """
+      Removes all the empty string in the nested dictionary reconstructedDict.
+      @ In, reconstructedDict, dictionary,  nested dictionary
+      @ Out, leanReconstructedDict, dictionary, nested dictionary without trailing blanks
+    """
+    if not isinstance(leanDict, (dict, list)):
+        return leanDict
+    if isinstance(leanDict, list):
```

Comment:
Well I may be wrong I think the three cases can be reached. Let me reprint the whole method here. 
```
    if not isinstance(reconstructedDict,(dict,list)):   # case 1
      return reconstructedDict
    if isinstance(reconstructedDict, list): # case 2
      return [v for v in (self.cleanEmpty(v) for v in reconstructedDict) if v]
    return {k: v for k, v in ((k, self.cleanEmpty(v)) for k, v in reconstructedDict.items()) if v} # case 3
```
case 1: return the input variable `reconstructedDict`  with no action if `reconstructedDict` is this is neither a list or dict
case 2: return the cleaned up list 
case 3 return the cleaned up dictionary
Let's say case 2 is not really here because I have not used it for a list.